powerauth-push-server/src/main/java/io/getlime/push/service/InboxService.java

```diff
@@ -88,6 +88,15 @@ public GetInboxMessageDetailResponse fetchMessageDetail(String userId, String ap
         return inboxMessageConverter.convertResponse(messageEntity.get());
     }
 
+    @Transactional(readOnly=true)
+    public GetInboxMessageDetailResponse fetchMessageDetail(String messageId) throws InboxMessageNotFoundException, AppNotFoundException {
+        final Optional<InboxMessageEntity> messageEntity = inboxRepository.findById(messageId);
+        if (!messageEntity.isPresent()) {
+            throw new InboxMessageNotFoundException("Unable to fetch message: " + messageId + ".");
+        }
+        return inboxMessageConverter.convertResponse(messageEntity.get());
```

Comment:
Personally, I prefer `orElseThrow` approach.
```suggestion
        final InboxMessageEntity messageEntity = inboxRepository.findById(messageId)..orElseThrow(() ->
                new InboxMessageNotFoundException("Unable to fetch message: " + messageId + "."));
        return inboxMessageConverter.convertResponse(messageEntity);
```

Reply:
We currently do not use this in the code - I will leave it as is for now.